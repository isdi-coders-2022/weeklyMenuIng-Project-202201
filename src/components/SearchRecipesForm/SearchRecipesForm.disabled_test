import { render, screen } from "@testing-library/react";
import SearchRecipesForm from "./SearchRecipesForm";
import TestRenderer from "react-test-renderer";

describe("Given a SearchRecipesForm Component", () => {
  describe("When it's rendered", () => {
    test("Then a form should be displayed", () => {
      render(<SearchRecipesForm />);

      const formInput = screen.queryByLabelText("Ingredients:");

      expect(formInput).toBeInTheDocument();
    });

    test("Then a form should display 6 checkboxes", () => {
      const expectedNumberOfCheckboxes = 6;
      render(<SearchRecipesForm />);

      const numOfCheckboxes = screen.queryAllByRole("checkbox").length;

      expect(numOfCheckboxes).toBe(expectedNumberOfCheckboxes);
    });
  });

  test("Then it should match the snapshot", () => {
    const searchRecipesForm = TestRenderer.create(
      <SearchRecipesForm />
    ).toJSON();

    expect(searchRecipesForm).toMatchSnapshot();
  });

  describe("submit button", () => {
    test("is disabled when instantiaded with empty text input", () => {
      render(<SearchRecipesForm />);

      expect(screen.getByText("Search recipes")).toBeDisabled();
    });
  });
});
