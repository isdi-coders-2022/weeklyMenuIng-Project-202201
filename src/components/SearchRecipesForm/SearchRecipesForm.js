import { useState } from "react";

const SearchRecipesForm = () => {
  const blankFields = {
    ingredients: "",
    "no-sugar": false,
    "high-protein": false,
    balanced: false,
    "low-sodium": false,
    vegan: false,
    vegetarian: false,
  };

  const [formData, setFormData] = useState(blankFields);

  const changeData = (event) => {
    setFormData({
      ...formData,
      [event.target.id]:
        event.target.type === "checkbox"
          ? event.target.checked
          : event.target.value,
    });
  };

  const filterResults = (data) => data;

  const onFormSubmit = (event) => {
    event.preventDefault();
    if (isFilled) {
      filterResults(formData);
      resetForm();
    }
  };

  const resetForm = () => {
    setFormData(blankFields);
  };

  const isFilled = formData.ingredients !== "";

  return (
    <form
      className="new-recipe"
      onSubmit={onFormSubmit}
      autoComplete="off"
      noValidate
    >
      <div className="form-group">
        <label htmlFor="ingredients">Ingredients: </label>
        <input
          type="text"
          id="ingredients"
          className="form-input"
          value={formData.ingredients}
          onChange={changeData}
        />
      </div>
      <div className="form-group">
        <input
          type="checkbox"
          id="no-sugar"
          checked={formData["no-sugar"]}
          onChange={changeData}
        />
        <label htmlFor="no-sugar">No-sugar</label>
      </div>
      <div className="form-group">
        <input
          type="checkbox"
          id="high-protein"
          checked={formData["high-protein"]}
          onChange={changeData}
        />
        <label htmlFor="high-protein">High-protein</label>
      </div>
      <div className="form-group">
        <input
          type="checkbox"
          id="balanced"
          checked={formData.balanced}
          onChange={changeData}
        />
        <label htmlFor="balanced">Balanced</label>
      </div>
      <div className="form-group">
        <input
          type="checkbox"
          id="low-sodium"
          checked={formData["low-sodium"]}
          onChange={changeData}
        />
        <label htmlFor="low-sodium">Low-sodium</label>
      </div>
      <div className="form-group">
        <input
          type="checkbox"
          id="vegan"
          checked={formData.vegan}
          onChange={changeData}
        />
        <label htmlFor="vegan">Vegan</label>
      </div>
      <div className="form-group">
        <input
          type="checkbox"
          id="vegetarian"
          checked={formData.vegetarian}
          onChange={changeData}
        />
        <label htmlFor="vegan">Vegetarian</label>
      </div>
      <button type="submit" disabled={!isFilled}>
        Search recipes
      </button>
    </form>
  );
};

export default SearchRecipesForm;
