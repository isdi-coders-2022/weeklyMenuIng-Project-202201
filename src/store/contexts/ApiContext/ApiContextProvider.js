import { useReducer } from "react";
import apiReducer from "../../reducers/api/apiReducer";
import ApiContext from "./ApiContext";

const ApiContextProvider = ({ children }) => {
  const [apiState, dispatcher] = useReducer(apiReducer, {
    isLoaded: true,
    hasError: false,
  });
  const data = { apiState, dispatcher };

  return <ApiContext.Provider value={data}>{children}</ApiContext.Provider>;
};

export default ApiContextProvider;
