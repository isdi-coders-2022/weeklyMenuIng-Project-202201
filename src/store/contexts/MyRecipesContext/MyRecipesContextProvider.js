import MyRecipesContext from "./MyRecipesContext";
import { useReducer } from "react";
import MyRecipesReducer from "../../reducers/myRecypes/myRcipesReducer";

const MyRecipesContextProvider = ({ children }) => {
  const [myRecipes, dispatch] = useReducer(MyRecipesReducer, []);
  return (
    <MyRecipesContext.Provider
      value={{
        myRecipes,
        dispatch,
      }}
    >
      {children}
    </MyRecipesContext.Provider>
  );
};
export default MyRecipesContextProvider;
