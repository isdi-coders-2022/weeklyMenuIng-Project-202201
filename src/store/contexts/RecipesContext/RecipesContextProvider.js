import RecipesContext from "./RecipesContext";
import { useReducer, useState } from "react";
import recipesReducer from "../../reducers/recipes/recipesReducer";
import recipeReducer from "../../reducers/recipes/recipeReducer";

const RecipesContextProvider = ({ children }) => {
  const [recipes, dispatch] = useReducer(recipesReducer, []);
  const [recipe, recipeDispatch] = useReducer(recipeReducer, {});

  const [nextEndPoint, setNextEndpoint] = useState("");
  return (
    <RecipesContext.Provider
      value={{
        recipes,
        dispatch,
        nextEndPoint,
        setNextEndpoint,
        recipe,
        recipeDispatch,
      }}
    >
      {children}
    </RecipesContext.Provider>
  );
};

export default RecipesContextProvider;
