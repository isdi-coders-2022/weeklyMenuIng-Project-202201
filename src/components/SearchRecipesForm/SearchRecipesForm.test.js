import { render, screen } from "@testing-library/react";
import SearchRecipesForm from "./SearchRecipesForm";
import ApiContextProvider from "../../store/contexts/ApiContext/ApiContextProvider";
import RecipesContextProvider from "../../store/contexts/RecipesContext/RecipesContextProvider";
import MyRecipesContextProvider from "../../store/contexts/MyRecipesContext/MyRecipesContextProvider";

describe("Given a SearchRecipesForm Component", () => {
  describe("When it's rendered", () => {
    test("Then a form should be displayed", () => {
      render(
        <ApiContextProvider>
          <MyRecipesContextProvider>
            <RecipesContextProvider>
              <SearchRecipesForm />
            </RecipesContextProvider>
          </MyRecipesContextProvider>
        </ApiContextProvider>
      );
      const formInput = screen.queryByPlaceholderText(
        "write some ingredients..."
      );

      expect(formInput).toBeInTheDocument();
    });

    test("Then a form should display 6 checkboxes", () => {
      const expectedNumberOfCheckboxes = 6;
      render(
        <ApiContextProvider>
          <MyRecipesContextProvider>
            <RecipesContextProvider>
              <SearchRecipesForm />
            </RecipesContextProvider>
          </MyRecipesContextProvider>
        </ApiContextProvider>
      );

      const numOfCheckboxes = screen.queryAllByRole("checkbox").length;

      expect(numOfCheckboxes).toBe(expectedNumberOfCheckboxes);
    });
  });

  describe("submit button", () => {
    test("is disabled when instantiaded with empty text input", () => {
      render(
        <ApiContextProvider>
          <MyRecipesContextProvider>
            <RecipesContextProvider>
              <SearchRecipesForm />
            </RecipesContextProvider>
          </MyRecipesContextProvider>
        </ApiContextProvider>
      );

      expect(screen.getByText("Search recipes")).toBeDisabled();
    });
  });
});
