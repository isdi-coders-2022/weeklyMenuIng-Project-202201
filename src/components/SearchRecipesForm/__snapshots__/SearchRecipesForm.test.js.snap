// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a SearchRecipesForm Component When it's rendered Then it should match the snapshot 1`] = `
<form
  autoComplete="off"
  className="search-recipes-form"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="ingredients-label"
      htmlFor="ingredients"
    >
      Ingredients:
    </label>
    <input
      className="form-input-ingredients"
      id="ingredients"
      onChange={[Function]}
      placeholder="write some ingredients..."
      type="text"
      value=""
    />
  </div>
  <label
    className="form-control"
    htmlFor="no-sugar"
  >
    <input
      checked={false}
      className="diets-checkbox"
      id="no-sugar"
      onChange={[Function]}
      type="checkbox"
    />
    no-sugar
  </label>
  <label
    className="form-control"
    htmlFor="high-protein"
  >
    <input
      checked={false}
      className="diets-checkbox"
      id="high-protein"
      onChange={[Function]}
      type="checkbox"
    />
    high-protein
  </label>
  <label
    className="form-control"
    htmlFor="balanced"
  >
    <input
      checked={false}
      className="diets-checkbox"
      id="balanced"
      onChange={[Function]}
      type="checkbox"
    />
    balanced
  </label>
  <label
    className="form-control"
    htmlFor="low-sodium"
  >
    <input
      checked={false}
      className="diets-checkbox"
      id="low-sodium"
      onChange={[Function]}
      type="checkbox"
    />
    low-sodium
  </label>
  <label
    className="form-control"
    htmlFor="vegan"
  >
    <input
      checked={false}
      className="diets-checkbox"
      id="vegan"
      onChange={[Function]}
      type="checkbox"
    />
    vegan
  </label>
  <label
    className="form-control"
    htmlFor="vegetarian"
  >
    <input
      checked={false}
      className="diets-checkbox"
      id="vegetarian"
      onChange={[Function]}
      type="checkbox"
    />
    vegetarian
  </label>
  <button
    disabled={true}
    type="submit"
  >
    Search recipes
  </button>
</form>
`;
